crates/items/src/item.rs

```diff
@@ -8854,3 +8855,24 @@ impl Item {
         }
     }
 }
+use std::convert::TryFrom;
+
+impl TryFrom<String> for Item {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        if let Some(item) = Item::from_name(value.as_str()) {
+            Ok(item)
+        } else {
+            Err(String::from("Unknown item name."))
```

Comment:
Use `&'static str` as the error type.

Reply:
Just being curious, but why should I use `&'static str` over `String`. Because of the static lifetime? I'm quite new to rust's lifetime system.

Reply:
`&'a str` is a reference to a `str`, which requires no allocations. `String` is an allocated `str` and can grow and expand like a `Vec`. In the case of `&'static str` we are referencing a `str` which is owned by the program, embedded literal, since it is owned by the program it has the lifetime `'static` which means it will be valid for the entire lifecycle of the program. We prefer `&'a str` over strings and `&'static str` is just a version which is valid for the entire program.

Reply:
Nice! Thanks :smile: